src/JsonmlToReact.js

```diff
@@ -36,26 +36,19 @@ export default class JsonmlToReact {
       return node;
     }
 
-    let attrs = Object.assign({ key: index }, JsonML.getAttributes(node));
-
-    if (attrs.class) {
-      attrs.className = attrs.class;
-      attrs.class = undefined;
-    }
-
-    if (attrs.style) {
-      attrs.style = toStyleObject(attrs.style);
-    }
-
+    const rawAttrs = Object.assign({}, JsonML.getAttributes(node));
     const tag = JsonML.getTagName(node);
     const converter = this.converters[tag];
-    const result = isFunction(converter) ? converter(attrs, data) : {};
+    const result = isFunction(converter) ? converter(rawAttrs, data) : {};
 
     const type = result.type || tag;
-    const props = result.props || attrs;
-
-    // reassign key in case `converter` removed it
-    props.key = props.key || index;
+    const resultProps = result.props || rawAttrs || {};
+    const props = Object.assign({}, resultProps, {
+      className: resultProps.className || resultProps.class,
+      class: undefined,
+      key: index,
+      style: rawAttrs.style && toStyleObject(rawAttrs.style)
+    });
```

Comment:
v.minor - this potentially adds an enumerable `style: undefined` to the object. i'm not sure if we care though.